### PR TITLE
test: add smoke test for user role for userdata crud api

### PR DESCRIPTION
**Pull request by shwstppr**
### Description

Adds smoke test for CRUD operations as a normal user

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by harikrishna-patnala** [comment]
code looks good to me, I'll wait for the smoke test results.

**Comment by shwstppr** [comment]
@blueorangutan test

File: test/integration/smoke/test_register_userdata.py [hunk 1]
```diff
@@ -762,3 +762,69 @@ def test_deploy_vm_with_registered_userdata_with_override_policy_deny(self):
...
+            self.template.userdataid,
+            None,
+            "Check userdata id in template response is None"
+        )
+
+        UserData.delete(
+            self.userapiclient,
+            id=self.userdata.userdata.id
+        )
+        self.cleanup.pop()
```
Comment (DaanHoogland) [inline test/integration/smoke/test_register_userdata.py:830]: :D I would have
```suggestion
        self.cleanup.remove(self.userdata)
```
but this should work as well :+1:

**Review by harikrishna-patnala — APPROVED** [review]
Verified the test results.
Test following operations as a normal user: ... === TestName: test_user_userdata_crud | Status : SUCCESS ===

LGTM

**Comment by shwstppr** [comment]
@blueorangutan package

**Review by rohityadavcloud — APPROVED** [review]
Thanks @shwstppr

**Comment by shwstppr** [comment]
@blueorangutan test

**Review by DaanHoogland — APPROVED** [review]

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by rohityadavcloud** [comment]
Merging this for 4.18, main was previously 4.18 when the smoketests had passed.

**Pull request merged**